noether_tpp/core/include/noether_tpp/core/types.h

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file types.h
+ * @copyright Copyright (c) 2021, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <vector>
+
+#include <Eigen/Geometry>
+
+namespace noether_tpp
+{
+/** @brief A point on a surface that makes up part of a path */
+using ToolPathWaypoint = Eigen::Isometry3d;
+
+/** @brief A set of contiguous waypoints that lie on the same line created by a "slice" through a
+ * surface */
+using ToolPathSegment = std::vector<Eigen::Isometry3d, Eigen::aligned_allocator<Eigen::Isometry3d>>;
+
+/** @brief A set of tool path segments that lie on the same line created by a tool path "slice",
+ * but are not connected (e.g. due to a hole in the mesh) */
+using ToolPath = std::vector<ToolPathSegment>;
+
+/** @brief A set of tool paths (i.e. rasters) generated by various slices through a surface */
+using ToolPaths = std::vector<ToolPath>;
+
+}  // namespace noether_tpp
```

Comment:
@marip8 I think I mentioned to you before I would prefer we define real types instead of typedefs for these things. But I won't press the issue.

Reply:
I don't mind replacing the `ToolPaths` typedef with the actual type. Are you suggesting we get rid of the other typedefs as well?

Reply:
@Levi-Armstrong can chime in on this, but I believe these typedefs exist in order to align our nomenclature directly with existing naming conventions in the CNC field.  Since we have a name defined, we could always upgrade from a typedef to a full class in the future, or change what it does in a minimally-invasive way.  Right now, Eigen types and arrays thereof meet our needs completely.  But these typedefs are useful in the same way that the below 'full name' class is more useful than std::vector<std::string>: The name tells you something about what you are doing.

```
class FullName
{
public:
  std::string first;
  std::string last;
  std::string suffix;
};

FullName n ("David", "Merz", "Jr");
std::vector<std::string> m = { "David", "Merz", "Jr" };
```

Reply:
Sorry for not being clear, I wasn't suggesting not defining names at all, I was suggesting it will be simpler if we start with a full class instead of a typedef. I suspect at least some of these objects will change into full classes eventually and it will be less breaking if they start that way. But again I won't press it any further.